botocore/regions.py

```diff
@@ -600,6 +609,14 @@ def _resolve_param_as_client_context_param(self, param_name):
             client_ctx_varname = client_ctx_params[param_name]
             return self._client_context.get(client_ctx_varname)
 
+    def _resolve_param_as_operation_context_param(
```

Comment:
We don't see to be handling default values when the following parameter is modeled:
```
"defaultStringParam": { 
    "type": "StringArray",
    "required": true,
    "default": ["A", "B", "C"] 
}
```
This seems like a miss.

Reply:
Good catch! Will add in the next commit.

Reply:
Defaults are handled elsewhere in the code.

When we are making a representation of the ruleset to attach to the client, we make a [parameter definition](https://github.com/boto/botocore/blob/b3cb3452225877f7fa823ab488d86bacd3211be9/botocore/endpoint_provider.py#L583) where we define a default if it is existed on the client.

This code is run later during endpoint resolution; we will try to evaluate that parameter.  If a default exists and we fail to resolve a value any other way, we will fall back to that default.  

While this PR does not require any changes to make the default functional, the testing bit below is a good callout.  I'll resolve this comment for now since it won't require a code change, but keep the comment about the testing until that is addressed.